server/webapp/WEB-INF/rails/webpack/single_page_apps/plugins.tsx

```diff
@@ -14,20 +14,35 @@
  * limitations under the License.
  */
 
-//fixme: can be deleted after components are created
+//fixme: this is the mount page which renders either old-plugins page or components-based plugins page based on the toggle
 
-const $              = require('jquery');
-const m              = require('mithril');
-const Stream         = require('mithril/stream');
-const PluginsWidget  = require('views/plugins/plugins_widget');
-const PluginInfos    = require('models/shared/plugin_infos');
-const PageLoadError  = require('views/shared/page_load_error');
+import Page from "../helpers/spa_base";
+import {PluginsPage} from "../views/pages/plugins";
+
+const $             = require('jquery');
+const m             = require('mithril');
+const Stream        = require('mithril/stream');
+const PluginsWidget = require('views/plugins/plugins_widget');
+const PluginInfos   = require('models/shared/plugin_infos');
+const PageLoadError = require('views/shared/page_load_error');
+
+export class PluginsSPA extends Page {
+  constructor() {
+    super(PluginsPage);
+  }
+}
 
 $(() => {
   const pluginElement = $('#plugins');
   const isUserAnAdmin = pluginElement.attr('is-user-an-admin');
 
-  const onSuccess = (pluginInfos) => {
+  //todo: This check determines whether the page needs to be rendered using old spa style or using new components styles
+  //pluginsElement div will only be available while rendering using old style.
+  if (pluginElement.get().length === 0) {
```

Comment:
There are two implementations of plugins page. One - old SPA and other - the new components style.
Depending on the toggle value, we need to render old/new page. To make it work (with minimal changes), this hack is required.

On the componentized plugins page, there will be no div with `plugins` id on it, hence it will render the components-plugins page.
Whereas, the old SPA will have a container div with `plugins` id.

This code will be removed soon once plugins SPA is componentized.
\cc: @akshaydewan @ketan